utils/webhook.js

```diff
@@ -0,0 +1,82 @@
+const { getResponseJSON, twilioSmsStatusTimeout } = require("./shared");
+const { processTwilioEvent, processSendGridEvent } = require("./firestore");
+const { SecretManagerServiceClient } = require("@google-cloud/secret-manager");
+const { EventWebhook, EventWebhookHeader } = require("@sendgrid/eventwebhook");
+const { getResponseJSON } = require("./shared");
+
+const sleep = (ms) => new Promise((res) => setTimeout(res, ms));
+
+/* This function will process the webhook data from Twilio */
+const receivedTwilioEvent = async (req, res) => {
+    try {
+        const status = req.body.MessageStatus;
+        // Delay to ensure twilio webhook event data returns in the correct order and ensure all messages have been sent and saved to the database
+        await sleep(twilioSmsStatusTimeout[status]);
```

Comment:
Waiting 10 seconds or longer before really handle the request is not optimal, in my opinion.  Better remove this delay, if possible.